### PR TITLE
Zkevm batches

### PR DESCRIPTION
**Pull request by isstuev**
resolves https://github.com/blockscout/frontend/issues/1122

ENV changes:

- NEXT_PUBLIC_IS_L2_NETWORK for Optimistic L2 networks was replaced with NEXT_PUBLIC_IS_OPTIMISTIC_L2_NETWORK
- NEXT_PUBLIC_L2_WITHDRAWAL_URL for Optimistic L2 networks was replaced with NEXT_PUBLIC_OPTIMISTIC_L2_WITHDRAWAL_URL
- NEXT_PUBLIC_IS_ZKEVM_L2_NETWORK was added for ZkEvm L2 networks

### CODE REVIEW

File: ui/zkEvmL2TxnBatches/ZkEvmL2TxnBatchDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    return <DataFetchAlert/>;
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  const sectionGap = (
```
Comment (tom2drum) [inline ui/zkEvmL2TxnBatches/ZkEvmL2TxnBatchDetails.tsx:61]: Better to use this component then - `ui/shared/DetailsInfoItemDivider.tsx`

File: ui/home/LatestZkEvmL2Batches.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { Box, Heading, Flex, Text, VStack } from '@chakra-ui/react';
+// import { AnimatePresence } from 'framer-motion';
```
[the remark this answers is not in the document]
Comment (isstuev) [reply ui/home/LatestZkEvmL2Batches.tsx:2]: I will add socket here later

File: ui/home/LatestZkEvmL2Batches.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          { dataToShow.map(((batch, index) => (
+            <LatestZkevmL2BatchItem
+              key={ batch.number + (isPlaceholderData ? String(index) : '') }
+              batch={ batch }
+              isLoading={ isPlaceholderData }
+            />
+          ))) }
+          { /* </AnimatePresence> */ }
+        </VStack>
+        <Flex justifyContent="center">
```
[the remark this answers is not in the document]
Comment (isstuev) [reply ui/home/LatestZkEvmL2Batches.tsx:47]: To center, as in Latest Blocks and Transactions

File: ui/shared/entities/address/__screenshots__/AddressEntity.pw.tsx_default_external-link-1.png
[the remark this answers is not in the document]
Comment (isstuev) [reply ui/shared/entities/address/__screenshots__/AddressEntity.pw.tsx_default_external-link-1.png:1]: I don't think external link should always have SM size, it should be changeable with props.

File: ui/shared/verificationSteps/VerificationStep.tsx
[the remark this answers is not in the document]
Comment (isstuev) [reply ui/shared/verificationSteps/VerificationStep.tsx:1]: but the component itself has nothing connected with ZK, it's just a sequence of steps with the particular icons and colors.

File: ui/shared/entities/block/ZkEvmBatchEntityL2.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { route } from 'nextjs-routes';
+
+import config from 'configs/app';
+import txBatchIcon from 'icons/txn_batches_slim.svg';
+
+import * as BlockEntity from './BlockEntity';
+
+const feature = config.features.zkEvmRollup;
+
+const BlockEntityL2 = (props: BlockEntity.EntityProps) => {
```
[the remark this answers is not in the document]
Comment (isstuev) [reply ui/shared/entities/block/ZkEvmBatchEntityL2.tsx:14]: but a batch isn't a block. As I understand, the architecture differs here (but I don't know why we use batch icon for a block for optimistic rollups)

File: ui/shared/entities/address/__screenshots__/AddressEntity.pw.tsx_default_external-link-1.png
[the remark this answers is not in the document]
Comment (tom2drum) [reply ui/shared/entities/address/__screenshots__/AddressEntity.pw.tsx_default_external-link-1.png:1]: ok, but can you check screenshots then. some of them should not been updated, like Token and TokenInstanceMetadata. the smaller font size and line height should been used for links in this components

**Review by tom2drum — APPROVED** [review]
ok but some screenshots are off

File: icons/unfinalized.svg
Comment (tom2drum) [inline icons/unfinalized.svg:1]: should it be something inside the box?) or mock-ups are not updated?
image

**Review by tom2drum — APPROVED** [review]

**Pull request merged**